jdaviz/configs/imviz/helper.py

```diff
@@ -241,14 +241,36 @@ def load_data(self, data, data_label=None, do_link=True, show_in_viewer=True, **
                                                     "be removed in a future release.  Use with "
                                                     "viz.batch_load() instead."))
 
-    def link_data(self, **kwargs):
+    def link_data(self, link_type='pixels', wcs_fallback_scheme=None, wcs_use_affine=True):
         """(Re)link loaded data in Imviz with the desired link type.
         All existing links will be replaced.
 
-        See :func:`~jdaviz.configs.imviz.helper.link_image_data`
-        for available keyword options and more details.
+        Parameters
+        ----------
+        link_type : {'pixels', 'wcs'}
+            Choose to link by pixels or WCS.
+
+        wcs_fallback_scheme : {None, 'pixels'}
+            If WCS linking failed, choose to fall back to linking by pixels or not at all.
+            This is only used when ``link_type='wcs'``.
+            Choosing `None` may result in some Imviz functionality not working properly.
+
+        wcs_use_affine : bool
+            Use an affine transform to represent the offset between images if possible
+            (requires that the approximation is accurate to within 1 pixel with the
+            full WCS transformations). If approximation fails, it will automatically
+            fall back to full WCS transformation. This is only used when ``link_type='wcs'``.
+            Affine approximation is much more performant at the cost of accuracy.
+
         """
-        link_image_data(self.app, **kwargs)
+        from jdaviz.configs.imviz.plugins.orientation.orientation import link_type_msg_to_trait
+        plg = self.plugins["Orientation"]
+        plg._obj.linking_in_progress = True
+        plg.link_type = link_type_msg_to_trait[link_type]
+        plg._obj.wcs_use_fallback = wcs_fallback_scheme == 'pixels'
+        plg.wcs_use_affine = wcs_use_affine
+        plg._obj.linking_in_progress = False
+        plg._obj._update_link()
```

Comment:
I'm not sure I understand this change. `orientation._update_link` calls `orientation._link_image_data` which calls `link_image_data`. So it looks like this is a longer and less clear way of doing a similar thing?

Reply:
I could not get `link_image_data` to work properly on its own when switching from pixel to WCS the first time. The viewer got stuck displaying the fake layer even though visibility is set properly when I check the viewer state. I think there is some magic the plugin is doing that the API alone isn't, and it is very hard to trace with the circular callbacks and async event handling. In the end, I convinced myself that this way makes the most sense to make `imviz.link_data(link_type="wcs")` to work properly again as advertised in the notebook.

Reply:
Can/should we deprecate `imviz.link_data` and schedule consolidating the logic (not necessarily here)?

Reply:
I was able to reproduce the behavior you're referring to, where the WCS-only layer is visible when calling `link_image_data `, by reverting the highlighted lines above. I'm a bit concerned that this does not work in this PR, but it does work in the main PR – what caused the change in this PR? Investigating a bit.